pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -123,12 +124,25 @@ func (r *ReconcilePolicy) addFinalizerAndUpdate(virtualMachine *kubevirt.Virtual
 	}
 
 	logger.V(1).Info("The VM does not have a finalizer")
-	virtualMachine.ObjectMeta.Finalizers = append(virtualMachine.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName)
 
-	err := r.Update(context.Background(), virtualMachine)
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// refresh vm instance
+		err := r.Get(context.TODO(), request.NamespacedName, virtualMachine)
+		if err != nil {
+			return errors.Wrap(err, "Failed to refresh vm instance")
+		}
+		virtualMachine.ObjectMeta.Finalizers = append(virtualMachine.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName)
+
+		err = r.Update(context.Background(), virtualMachine)
+		if err != nil {
+			return errors.Wrap(err, "failed to update the VM with the new finalizer")
```

Comment:
There is a bug here, when RetryOnConflict is used, you have to return the error exactly at it was produced so it can check it out.

Reply:
will be fixed in another PR